gears/src/rest/tendermint_events_handler.rs

```diff
@@ -0,0 +1,33 @@
+// TODO: update errors
+use anyhow::anyhow;
+
+pub struct StrEventsHandler<'a> {
+    events_str: &'a str,
+}
+
+impl<'a> StrEventsHandler<'a> {
+    pub fn new(events_str: &'a str) -> Self {
+        let events_str = events_str.trim_start_matches('\'').trim_end_matches('\'');
+        Self { events_str }
+    }
+
+    pub fn try_parse_tendermint_events_vec(&self) -> anyhow::Result<Vec<String>> {
+        let events = self.events_str.split('&');
+
+        let mut tm_events = Vec::with_capacity(self.events_str.matches('&').count() + 1);
+        for event in events {
+            if !event.contains('=') {
+                return Err(anyhow!("invalid event; event {event} should be of the format: {{eventType}}.{{eventAttribute}}={{value}}"));
+            }
+            let tokens = event
+                .split_once('=')
+                .expect("the condition is checked above");
+            if tokens.0 == "tx.height" {
```

Comment:
Should every `value` which is number follow this rule?

Reply:
With smth like this: https://doc.rust-lang.org/stable/std/primitive.char.html#method.is_numeric

Reply:
It shouldn't follow the rule. The query has only one exception for `tx.height` event type.

Reply:
I should have guessed that this is weird requirement from cosmos.